pkg/controller/user/user_controller.go

```diff
@@ -531,6 +570,74 @@ func (c *Controller) deleteRoleBindings(user *iamv1alpha2.User) error {
 	return nil
 }
 
+func (c *Controller) deleteLoginRecords(user *iamv1alpha2.User) error {
+	listOptions := metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labels.Set{iamv1alpha2.UserReferenceLabel: user.Name}).String(),
+	}
+	deleteOptions := metav1.NewDeleteOptions(0)
+
+	if err := c.ksClient.IamV1alpha2().LoginRecords().
+		DeleteCollection(deleteOptions, listOptions); err != nil {
+		klog.Error(err)
+		return err
+	}
+	return nil
+}
+
+func (c *Controller) syncUserStatus(user *iamv1alpha2.User) (*iamv1alpha2.User, error) {
+	records, err := c.loginRecordLister.List(labels.SelectorFromSet(labels.Set{iamv1alpha2.UserReferenceLabel: user.Name}))
+	if err != nil {
+		klog.Error(err)
+		return nil, err
+	}
+	now := time.Now()
+	attempts := 0
+	for _, record := range records {
```

Comment:
We'd reorganize this part by user.state
1. user.state == disabled -> return
2. user.state == blocked -> blocked duration exceeds maximum duration -> unblock -> return
3. user.state == active -> list failed records -> block if exceeds maximum retries -> return